spock-core/src/main/java/org/spockframework/runtime/PlatformSpecRunner.java

```diff
@@ -320,7 +325,16 @@ void runFeatureMethod(SpockExecutionContext context) {
 
     MethodInfo featureIteration = new MethodInfo(context.getCurrentFeature().getFeatureMethod());
     featureIteration.setIteration(context.getCurrentIteration());
-    invoke(context, context.getCurrentInstance(), featureIteration, context.getCurrentIteration().getDataValues());
+
+    Class<?>[] parameterTypes = featureIteration.getReflection().getParameterTypes();
+    int parameterCount = parameterTypes.length;
+    Object[] dataValues = context.getCurrentIteration().getDataValues();
+
+    Object[] iterationArguments = new Object[parameterCount];
+    arraycopy(dataValues, 0, iterationArguments, 0, dataValues.length);
+    Arrays.fill(iterationArguments, dataValues.length, parameterCount, MISSING_ARGUMENT);
```

Comment:
You mean instead of `MISSING_ARGUMENT`?
We could surely do that if you think it makes sense.
But yeah, it might be better to do this in a followup task maybe?

Reply:
Btw., is the exception I throw adequate?